### ISSUE TITLE
gh-pages setup

### ISSUE THREAD

**Issue by imlucas** [mongodb-js/mongodb-js.github.io#1]
From @medina:

> I'd absolutely love for us to have a better view into all of these projects, e.g., what http://netflix.github.io is for https://github.com/Netflix.  Perhaps the front-end-inclined could assist with that? :-)

I've been thinking about this quite a bit as well.  Making sure tests are passing once, pull requests aren't languishing, something unexpectedly becoming very popular, etc is really hard once you have > 20 live modules.  (see [mongoscope/modules.md](https://github.com/10gen/mongoscope/blob/dev/docs/modules.md)) So how about we try 2 birds with 1 stone?  Some things to make this happen:
- [x] project setup (usual toolchain)
- [x] <strike>ping one of the npm search api's for all modules (eg keyword mongodb.js) or make a little heroku server endpoint</strike> hit github api for /repos (pull over ampersand models and collections from dyno)
- [x] <strike>ideas for module avatars?</strike> configurable + default to http://robohash.org/
- [x] sort repos by newest, # open issues, latest update, most popular, open pull requests
- [x] popovers for grid layout
- [x] fixed width list items in list layout
## examples
- [twitter](http://twitter.github.io/) [src](https://github.com/twitter/twitter.github.com)
- [netflix](http://netflix.github.io/) [src](https://github.com/Netflix/netflix.github.com)

**Comment by medina** [comment]
An example closer to home for the repo view: [mongodb-tools.com](http://mongodb-tools.com) [[src](https://github.com/IanWhalen/mongodb-tools.com)]

As an example of a "health check" that's less related to a build status, see 10gen/corp#294 for something that checks whether a PR submitter has signed our contributor agreement (but probably out of scope here).

**Comment by imlucas** [comment]
@medina Ah so something a little more like a faceted search?  Other categories/sorts you can think of?  

![org page](https://cloud.githubusercontent.com/assets/23074/6491500/8d6b4fd4-c27b-11e4-9919-2a0c1c1ed0fb.png)

**Comment by imlucas** [comment]
Filled out list item views:

![org page 1](https://cloud.githubusercontent.com/assets/23074/6492057/79e05d52-c27f-11e4-9204-d423478384c9.png)

**Comment by medina** [comment]
Nice.

Would we be able to add our own layered meta-data to what's available in the GitHub API, like knowing the "support status" of a project ("labs" vs "commercial support available")?

**Comment by imlucas** [comment]
@medina yeah I was thinking about this for how to support categories (could be any arbitrary tagging though):

``` javascript
defineCategories([
  'Driver', function(repo){
    return /driver/i.test(repo.name + ' ' + repo.description; 
  },
  'Database': function(repo){
    return repo.name === 'mongo';
  }
]);
```

**Comment by imlucas** [comment]
@medina think I have just enough to start hacking up a prototype.  The most important question though remains unanswered: what about avatars for repos? :)

google's color block + first initial wont do because almost all start with M.  Audubon pencil drawings?  Sega genesis characters from 1997?

**Comment by medina** [comment]
Option to source a repo-specific avatar file if it's available, [RoboHash](http://robohash.org) if none?

**Comment by imlucas** [comment]
Ah excellent!  All the setup is done.  Just need to finish wiring in the morning.

**Comment by imlucas** [comment]
Progress.  Next up: make facet sidebar, sorting, grid view work.

![](https://cldup.com/fGk2KGXuGf.png)

**Comment by imlucas** [comment]
Grid view
![screen shot 2015-03-05 at 4 18 04 pm](https://cloud.githubusercontent.com/assets/23074/6514804/43f6c95e-c353-11e4-9b24-d604f268a7cc.png)

**Comment by imlucas** [comment]
Working.  Now for better dynamic config and makin pretty.  Missing anything @medina ?

http://mongodb-org-page.surge.sh/#repos/stargazers_count/grid

**Comment by imlucas** [comment]
Going to sleep on this.

![screen shot 2015-03-06 at 12 16 14 am](https://cloud.githubusercontent.com/assets/23074/6520563/1822f06c-c396-11e4-9fb1-fa4e66d4e060.png)
![screen shot 2015-03-06 at 12 16 35 am](https://cloud.githubusercontent.com/assets/23074/6520564/182802e6-c396-11e4-8c84-4d902b328344.png)

**Comment by imlucas** [comment]
Little polish.  Bot animation on hover, popovers for grid view.

![screen shot 2015-03-07 at 1 15 06 am](https://cloud.githubusercontent.com/assets/23074/6540358/ac41556e-c467-11e4-9b17-4c729fa06c65.png)

**Comment by imlucas** [comment]
mongodb org page available after #15
